src/js/components/user/connect.tsx

```diff
@@ -133,28 +91,69 @@ const ConnectModal = ({
             key="submit"
             label={t('Continue')}
             variant="brand"
-            onClick={handleCustomDomainConnect}
+            onClick={submitClicked}
+            disabled={!customDomain}
           />,
         ]
       }
       onRequestClose={handleClose}
     >
       {isCustomDomain ? (
-        <CustomDomainForm
-          url={url}
-          setUrl={setUrl}
-          handleCustomDomainConnect={handleCustomDomainConnect}
-        />
+        /* POSTing instead of redirecting to the login endpoint is more secure */
+        <form
+          className="slds-p-around_large"
+          action={window.api_urls.salesforce_login?.()}
+          method="POST"
+          onSubmit={handleCustomDomainConnect}
+        >
+          <input type="hidden" name="csrfmiddlewaretoken" value={csrftoken} />
+          <input type="hidden" name="next" value={window.location.pathname} />
+          <input type="hidden" name="custom_domain" value={customDomain} />
```

Comment:
@jerivas This is close, but I think the SF login handler needs to be adjusted to read this `custom_domain` from `POST` instead of `GET`: https://github.com/oddbird/Metecho/blob/main/metecho/oauth2/salesforce/views.py#L31-L33

Reply:
Fixed in 555899ab263d5b1e824f11b47644105af87e70b8